src/main/java/de/zalando/ep/zalenium/registry/ZaleniumRegistry.java

```diff
@@ -449,6 +450,31 @@ public void uncaughtException(Thread t, Throwable e) {
         }
     }
 
+    @Override
+    public HttpClient getHttpClient(URL url) {
+        // https://github.com/zalando/zalenium/issues/491
+        int maxTries = 3;
+        for (int i = 0; i < maxTries; i++) {
```

Comment:
Maybe `for (int i = 1; i <= maxTries; i++)` might work better here, so you can handle the exception throw easier.

Reply:
Right that is more simple.